### ISSUE TITLE
Add Sentry exception capture

### ISSUE THREAD

**Issue by jwhitlock** [mozilla-it/ctms-api#139]
Add a middleware to catch exceptions and report to [Sentry](https://sentry.prod.mozaws.net). Work with ops to add Sentry buckets for stage and production, and to add the team to them.

Part of Jira:OSS-333

**Comment by jwhitlock** [comment]
The sentry integration has been deployed. It was interfering with tests, which required moving initialization into the FastAPI setup (which doesn't run in tests), in PR #150.

@duallain added the Sentry configuration to environment settings, so everything should work. We didn't get any natural exceptions, so I forced the issue by getting a shell in staging, and running inside Python:

```python
from sentry_sdk import init, capture_message
init()
capture_message("I am a message from staging")
```

This message showed up in the Sentry project, so I'm more confident that exceptions will as well.

However, we got a handful of 500 errors from sentry stage to basket dev, without a Sentry traceback for CTMS. I'm digging in  to see if I can determine the issue.
